core/iwasm/aot/aot_loader.c

```diff
@@ -1173,12 +1181,14 @@ load_types(const uint8 **p_buf, const uint8 *buf_end, AOTModule *module,
 
     /* Create each function type */
     for (i = 0; i < module->type_count; i++) {
+        uint32 type_flag;
         uint32 param_count, result_count;
         uint32 param_cell_num, ret_cell_num;
         uint64 size1;
 
-        read_uint32(buf, buf_end, param_count);
-        read_uint32(buf, buf_end, result_count);
+        read_uint32(buf, buf_end, type_flag);
+        read_uint16(buf, buf_end, param_count);
+        read_uint16(buf, buf_end, result_count);
```

Comment:
It is good to use uint16 since it is defined as uint16 in WASMType struct, let's summarize the changes and modify the document later.